### DISCUSSION TITLE
Use local client browser storage instead of centrally storing the chats

### DISCUSSION THREAD

**Discussion by awaescher** [open-webui/open-webui#4082] [Ideas]
I would love have no chats stored centrally and instead having open-webui using the local browser storage of every client exclusively. That means that only the browser keeps the chat history and noone else has access to it. Not even the same user on another machine or even on another browser on the same machine.

[Chatbot-Ollama](https://github.com/ivanfioravanti/chatbot-ollama) does this which makes it super easy to set up and even easier for the users to make use of it as there is **no user administration required**.

However, everything else about open-webui is better than Chatbot-Ollama. 

One thing to note here: This might cause that there's no distinction between admins and users anymore. Personally, I would love to define every administrative setting in a docker-compose.yml but we might need the admin area to be password protected or so for those who would prefer the admin UI.
 
---

This was an issue originally which was closed as "not planned": #4054 
I would love to hear if the community would love to have this feature.

**Comment by justinh-rahb** [comment]
It's been suggested before, and it would indeed be a useful feature for another that we someday want to reintroduce: the _Lite_ version of Open WebUI that has no backend. If we did that maybe we can also have a minimal backend option too (basically just an API proxy, no RAG, no running community code). Anyway I'll dig up some related issue/discussion numbers to link later.

**Reply by github123520** [reply]
I also need this feature: "Add a skip {SKIP} button to the user registration interface", in fact, I hate the function of registering an account, there are too many account passwords that need to be remembered on the network today, I hope to provide a docker version that does not require registration and login account, to provide a small number of users who "do not need account management", thank you very much.

**Reply by bioshazard** [reply]
Really should be offline-first but configurable with backend provider.

**Comment by bioshazard** [comment]
@justinh-rahb So I have built out a whole chat front end at my enterprise that does a basic form of much of the front end with dexie. I kinda want to abandon my whole project in favor of leaning into OpenWebUI, but "no centralized chats on server" is a major requirement. It also sets us up nicely to go serverless on oai compat  standard early next year. Would the maintainers of this project except such a contribution if I attempted it? Who specifically would need to sign off on the design and actual code? I am wary to attempt contribution without maintainer alignment. Happy to chat about this too lmk.

**Comment by JARZcorp** [comment]
I really like the idea especially in a GDPR compliant environment it would save countless discussions! This has not been picked up anymore though, right?

**Comment by bstrdsmkr** [comment]
Just as another data point, we have the opposite need -- we need to centrally maintain all chats for compliance reasons, so if implemented, please remember to make it optional :)

**Comment by surak** [comment]
I heard of some organization which forked open-webui and did exactly that - chats are stored as cookies, not in the database.

I asked them for access to their code, and if I have it, I will make a pull request. 

It's more than privacy - I absolutely don't want to have users' conversations clogging my database, and having to synchronize them among different kubernetes' instances etc

**Reply by QuantumNomad26** [reply]
Any new on this? No centralized chat database is one major requirement for my organization so i wont be able to use open web ui without it.

**Reply by nightscape** [reply]
@surak any news on this?

**Reply by surak** [reply]
they never replied :-(

**Reply by nightscape** [reply]
Ok, thanks for getting back!
I found LobeChat which (before the transition to LobeHub) seems to be able to store data in the browser as well.

**Comment by JensGM** [comment]
Any news on this? This would be great
